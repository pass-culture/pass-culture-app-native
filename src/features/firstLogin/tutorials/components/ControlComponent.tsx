import React from 'react'
import { TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

import { ArrowNext } from 'ui/svg/icons/ArrowNext'
import { ArrowPrevious } from 'ui/svg/icons/ArrowPrevious'

export type ControlComponentProps = {
  onPress: () => void
  title: string
  type: 'prev' | 'next'
}

const StyledTouchableOpacity = styled(TouchableOpacity)<{ type: string }>(({ type }) => ({
  [type === 'prev' ? 'marginLeft' : 'marginRight']: '50%',
  height: '100%',
  width: '100%',
  alignItems: 'center',
  justifyContent: 'center',
}))

export const ControlComponent = ({ onPress, title, type }: ControlComponentProps) => {
  return (
    <StyledTouchableOpacity
      accessibilityLabel={title}
      onPress={onPress}
      type={type}
      testID="button">
      {type === 'prev' ? (
        <ArrowPrevious testID="arrowPrevious" />
      ) : (
        <ArrowNext testID="arrowNext" />
      )}
    </StyledTouchableOpacity>
  )
}
