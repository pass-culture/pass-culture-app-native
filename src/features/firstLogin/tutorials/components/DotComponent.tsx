import React from 'react'
import { SwiperProps } from 'react-native-web-swiper'
import styled from 'styled-components/native'

import { Dot } from 'ui/svg/icons/Dot'
import { ColorsEnum, getSpacing } from 'ui/theme'

const CURRENT_STEP_SIZE = 12
const DEFAULT_SIZE = 8

type DotComponentProps = SwiperProps & {
  isActive?: boolean
}

// Class typing forced due to https://github.com/reactrondev/react-native-web-swiper/issues/56
export class DotComponent extends React.Component<DotComponentProps> {
  render() {
    const { isActive } = this.props
    return (
      <DotContainer>
        <Dot
          color={isActive ? ColorsEnum.PRIMARY : ColorsEnum.GREY_MEDIUM}
          size={isActive ? CURRENT_STEP_SIZE : DEFAULT_SIZE}
          testID="dot-icon"
        />
      </DotContainer>
    )
  }
}

const DotContainer = styled.View({
  alignSelf: 'center',
  marginLeft: getSpacing(2),
  marginRight: getSpacing(2),
})
