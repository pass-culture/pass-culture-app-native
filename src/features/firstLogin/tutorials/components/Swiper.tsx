import React from 'react'
import GestureRecognizer from 'react-native-swipe-gestures'

const CONFIG = {
  velocityThreshold: 0.03,
  directionalOffsetThreshold: 400,
  gestureIsClickThreshold: 0.1,
}

type Props = Omit<GestureRecognizer['props'], 'config'>

export const Swiper: React.FunctionComponent<Props> = (props) => {
  return (
    <GestureRecognizer {...props} config={CONFIG}>
      {props.children}
    </GestureRecognizer>
  )
}
