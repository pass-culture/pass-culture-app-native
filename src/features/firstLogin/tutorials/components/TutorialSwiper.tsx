import React from 'react'
import GestureRecognizer from 'react-native-swipe-gestures'
import styled from 'styled-components/native'

const CONFIG = {
  velocityThreshold: 0.03,
  directionalOffsetThreshold: 400,
  gestureIsClickThreshold: 0.1,
}

type Props = Omit<GestureRecognizer['props'], 'config'>

export const TutorialSwiper: React.FunctionComponent<Props> = (props) => {
  return (
    <StyledGestureRecognizer {...props} config={CONFIG}>
      {props.children}
    </StyledGestureRecognizer>
  )
}

const StyledGestureRecognizer = styled(GestureRecognizer)({
  flexGrow: 1,
})
