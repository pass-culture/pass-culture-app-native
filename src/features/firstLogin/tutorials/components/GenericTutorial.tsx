import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import LottieView from 'lottie-react-native'
import AnimatedLottieView from 'lottie-react-native'
import React, { FunctionComponent, useEffect, useRef } from 'react'
import styled from 'styled-components/native'

import { UseNavigationType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import { AnimationObject } from 'ui/animations/type'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { ButtonTertiaryGreyDark } from 'ui/components/buttons/ButtonTertiaryGreyDark'
import { StepDots } from 'ui/components/StepDots'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'

type Props = {
  animation: AnimationObject
  animationSize: number
  buttonCallback: () => void
  buttonText: string
  pauseAnimationOnRenderAtFrame?: number
  step: number
  subTitle: string
  text: string
  title: string
}

export const GenericTutorial: FunctionComponent<Props> = (props) => {
  const { navigate } = useNavigation<UseNavigationType>()

  const animationRef = useRef<AnimatedLottieView>(null)

  useEffect(() => {
    if (props.pauseAnimationOnRenderAtFrame) {
      animationRef.current?.play(0, props.pauseAnimationOnRenderAtFrame)
    }
  }, [])

  function goToHomeWithoutModal() {
    navigate('TabNavigator')
  }

  return (
    <Container>
      <Spacer.Flex flex={1} />
      <Header>
        <ButtonTertiaryGreyDark title={_(t`Tout passer`)} onPress={goToHomeWithoutModal} />
      </Header>
      <Spacer.Flex flex={2} />
      <StyledLottieView
        ref={animationRef}
        source={props.animation}
        loop={false}
        size={props.animationSize}
      />
      <Spacer.Flex flex={0.8} />
      <StyledTitle>{props.title}</StyledTitle>
      <StyledSubTitle>{props.subTitle}</StyledSubTitle>
      <Spacer.Flex flex={0.8} />
      <StyledBody>{props.text}</StyledBody>
      <Spacer.Flex flex={2} />
      <ButtonContainer>
        <ButtonPrimary title={props.buttonText} onPress={props.buttonCallback} />
      </ButtonContainer>
      <Spacer.Flex flex={0.8} />
      <StepDots numberOfSteps={4} currentStep={props.step} />
      <Spacer.Flex flex={1} />
    </Container>
  )
}

const Header = styled.View({
  alignSelf: 'flex-end',
  width: '40%',
})

const ButtonContainer = styled.View({
  flexDirection: 'row',
  width: '100%',
  paddingHorizontal: getSpacing(6),
  justifyContent: 'center',
})

const StyledLottieView = styled(LottieView)((props: { size: number }) => ({
  width: props.size,
  height: props.size,
}))

const StyledBody = styled(Typo.Body).attrs({
  color: ColorsEnum.BLACK,
})({
  width: '60%',
  textAlign: 'center',
  flexDirection: 'column',
})

const Container = styled.View({
  alignItems: 'center',
  flexGrow: 1,
})

const StyledTitle = styled(Typo.Title1).attrs({
  color: ColorsEnum.BLACK,
})({
  display: 'flex',
})

const StyledSubTitle = styled(Typo.Title2).attrs({
  color: ColorsEnum.BLACK,
})({
  textAlign: 'center',
})
