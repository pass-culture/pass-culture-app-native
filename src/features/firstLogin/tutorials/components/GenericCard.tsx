import AnimatedLottieView from 'lottie-react-native'
import LottieView from 'lottie-react-native'
import React, { FunctionComponent, RefObject, useEffect } from 'react'
import { View } from 'react-native'
import * as Animatable from 'react-native-animatable'
import Swiper from 'react-native-web-swiper'
import styled from 'styled-components/native'

import { analytics } from 'libs/analytics'
import { AnimationObject } from 'ui/animations/type'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { Spacer } from 'ui/components/spacer/Spacer'
import { getSpacing, Typo } from 'ui/theme'

export type CardKey = {
  activeIndex?: number
  index?: number
  key?: number
  name?: string
  swiperRef?: RefObject<Swiper>
}

export type CardProps = CardKey & {
  animation: AnimationObject
  buttonCallback?: () => void
  buttonText: string
  pauseAnimationOnRenderAtFrame: number
  subTitle: string
  text: string
  title: string
}

export let didFadeIn = false

export const GenericCard: FunctionComponent<CardProps> = (props: CardProps) => {
  const animationRef = React.useRef<AnimatedLottieView>(null)
  const animatedButtonRef = React.useRef<Animatable.View & View>(null)

  didFadeIn = false
  useEffect(() => {
    if (props.index === props.activeIndex) {
      animationRef.current?.play(0, props.pauseAnimationOnRenderAtFrame)
    } else {
      animationRef.current?.pause()
    }
  }, [props.pauseAnimationOnRenderAtFrame, animationRef, props.index, props.activeIndex])

  useEffect(() => {
    const button = animatedButtonRef?.current
    if (button?.fadeIn) {
      if (props.index === props.activeIndex) {
        button.fadeIn(1000)
        didFadeIn = true
      }
    }
  }, [animatedButtonRef])

  useEffect(() => {
    if (props.name && props.index !== undefined && props.activeIndex === props.index) {
      analytics.logScreenView(props.name)
    }
  }, [props.name, props.index, props.activeIndex])

  return (
    <GenericCardContainer>
      <Spacer.Flex flex={2} />
      <StyledLottieContainer>
        <StyledLottieView
          key={props.activeIndex}
          ref={animationRef}
          source={props.animation}
          loop={false}
          resizeMode="contain"
        />
      </StyledLottieContainer>
      <Spacer.Flex flex={1} />
      <StyledTitle>{props.title}</StyledTitle>
      <StyledSubTitle>{props.subTitle}</StyledSubTitle>
      <Spacer.Flex flex={1} />
      <StyledBody>{props.text}</StyledBody>
      <Spacer.Flex flex={2} />
      <Animatable.View ref={animatedButtonRef}>
        {props.activeIndex === props.index ? (
          <ButtonPrimary title={props.buttonText} onPress={props.buttonCallback} />
        ) : (
          <InvisibleButtonHeight testID="invisible-button-height" />
        )}
      </Animatable.View>
      <Spacer.Flex flex={3} />
    </GenericCardContainer>
  )
}

const InvisibleButtonHeight = styled.View({
  height: getSpacing(12),
})

const StyledLottieContainer = styled.View({
  flexGrow: 1,
  alignItems: 'center',
})

const GenericCardContainer = styled.View({
  flex: 1,
  paddingHorizontal: getSpacing(5),
})

const StyledLottieView = styled(LottieView)({
  width: getSpacing(60),
  height: getSpacing(60),
})

const StyledTitle = styled(Typo.Title1)({
  textAlign: 'center',
})

const StyledSubTitle = styled(Typo.Title2)({
  textAlign: 'center',
})

const StyledBody = styled(Typo.Body)({
  textAlign: 'center',
})
