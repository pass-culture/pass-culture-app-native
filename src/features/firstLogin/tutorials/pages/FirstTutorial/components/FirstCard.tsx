import { t } from '@lingui/macro'
import React from 'react'

import { CardKey, GenericCard } from 'features/firstLogin/tutorials/components/GenericCard'
import { _ } from 'libs/i18n'
import { storage } from 'libs/storage'
import TutorialPassLogo from 'ui/animations/tutorial_pass_logo.json'

export function FirstCard(props: CardKey) {
  function onButtonPress() {
    storage.saveObject('has_seen_tutorials', true)
    props.swiperRef?.current?.goToNext()
  }
  return (
    <GenericCard
      animation={TutorialPassLogo}
      buttonCallback={onButtonPress}
      buttonText={_(t`Continuer`)}
      pauseAnimationOnRenderAtFrame={62}
      subTitle={_(t`c'est...`)}
      text={_(t`une initiative financée par le Ministère de la Culture.`)}
      title={_(t`Le pass Culture`)}
      {...props}
    />
  )
}
