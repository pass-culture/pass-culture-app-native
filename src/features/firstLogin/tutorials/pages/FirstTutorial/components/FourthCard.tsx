import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React, { useEffect } from 'react'

import { CardKey, GenericCard } from 'features/firstLogin/tutorials/components/GenericCard'
import { UseNavigationType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import { storage } from 'libs/storage'
import StarAnimation from 'ui/animations/tutorial_star.json'

export function FourthCard(props: CardKey) {
  const { navigate } = useNavigation<UseNavigationType>()

  useEffect(() => {
    storage.saveObject('has_seen_tutorials', true)
  }, [])

  function onButtonPress() {
    navigate('TabNavigator')
  }

  return (
    <GenericCard
      animation={StarAnimation}
      buttonCallback={onButtonPress}
      buttonText={_(t`Découvrir`)}
      pauseAnimationOnRenderAtFrame={62}
      subTitle={_(t`quotidiennes`)}
      text={_(
        t`Nos nombreux partenaires ajoutent de nouvelles offres quotidiennement. Découvre les dès maintenant !`
      )}
      title={_(t`Des nouveautés`)}
      {...props}
    />
  )
}
