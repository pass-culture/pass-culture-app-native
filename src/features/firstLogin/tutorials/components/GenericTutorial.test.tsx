import AsyncStorage from '@react-native-community/async-storage'
import { act, fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import { Text, View } from 'react-native'

import { navigate } from '__mocks__/@react-navigation/native'
import { CardKey } from 'features/firstLogin/tutorials/components/GenericCard'
import { analytics } from 'libs/analytics'

import { GenericTutorial, Props } from './GenericTutorial'

describe('<GenericTutorial />', () => {
  const TestCard = (props: CardKey) => (
    <View>
      <Text>{props.swiperRef ? 'swipeRef exist' : 'swipeRef does not exist'}</Text>
      <Text>{props.name ? 'name exist' : 'name does not exist'}</Text>
    </View>
  )

  it('should render correctly', () => {
    const renderAPI = renderGenericTutorialComponent()
    expect(renderAPI).toMatchSnapshot()
  })

  it('should redirect on home, set has_seen_tutorials in storage and run analytics when pressing skip all button', async () => {
    const name = 'FirstTutorial'
    const { getByText } = renderGenericTutorialComponent({
      name: name,
      children: [<TestCard key={undefined} />],
    })
    const skipAll = await getByText('Tout passer')
    fireEvent.press(skipAll)
    expect(AsyncStorage.setItem).toBeCalledWith('has_seen_tutorials', 'true')
    expect(navigate).toHaveBeenCalledWith('TabNavigator')
    await act(async () => {
      expect(analytics.logHasSkippedTutorial).toHaveBeenCalledWith(`${name}1`)
    })
  })

  it('should have a swiperRef passed to each children', async () => {
    const { getByText } = renderGenericTutorialComponent({
      name: 'FirstTutorial',
      children: [<TestCard key={undefined} />],
    })
    expect(await getByText('swipeRef exist')).toBeTruthy()
    expect(() => getByText('swipeRef does not exist')).toThrow()
  })
  it('should have an automatically set name passed to each children', async () => {
    const { getByText } = renderGenericTutorialComponent({
      name: 'Tuto',
      children: [<TestCard key={undefined} />],
    })
    expect(await getByText('name exist')).toBeTruthy()
    expect(() => getByText('name does not exist')).toThrow()
  })
})

function renderGenericTutorialComponent(props: Props = { name: 'FirstTutorial', children: [] }) {
  return render(<GenericTutorial {...props} />)
}
