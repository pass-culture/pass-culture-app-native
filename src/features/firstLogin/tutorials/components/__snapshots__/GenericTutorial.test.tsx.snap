// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericTutorial /> should render correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      height="100%"
      width="100%"
    >
      undefined-SVG-Mock
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "display": "flex",
          "flexGrow": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      customHeight={16}
      style={
        Array [
          Object {
            "height": 16,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexGrow": 1,
            "maxHeight": 900,
            "maxWidth": 500,
            "paddingTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 24,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "maxWidth": 500,
                "opacity": 1,
                "paddingBottom": 2,
                "paddingLeft": 2,
                "paddingRight": 2,
                "paddingTop": 2,
                "width": "100%",
              }
            }
            testID="button-container"
          >
            <Text
              adjustsFontSizeToFit={false}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#626262",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 15,
                    "lineHeight": 20,
                    "marginLeft": 5,
                    "maxWidth": 650,
                  },
                ]
              }
              testID="button-title"
              textColor="#626262"
            >
              Tout passer
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "width": "100%",
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 0,
                "justifyContent": "space-between",
                "left": 0,
                "overflow": "hidden",
                "position": "relative",
                "top": 0,
                "width": 0,
              }
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Object {
                  "flexDirection": "row",
                  "height": 0,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 0,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 0,
                  "margin": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 0,
                  "margin": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 0,
                  "margin": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "minHeight": 1,
                        "minWidth": 1,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "height": 0,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "position": "absolute",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        numberOfSpaces={8}
        style={
          Array [
            Object {
              "height": 32,
            },
          ]
        }
      />
    </View>
    <View
      customHeight={8}
      style={
        Array [
          Object {
            "height": 8,
          },
        ]
      }
    />
  </View>,
]
`;
